client/src/main/java/io/mantisrx/server/master/client/MantisMasterClientApi.java

```diff
@@ -660,22 +679,43 @@ private Integer getSinkStageNumFromJsonResponse(String jobId, String response) t
                         .connect()
                         .flatMap((ObservableConnection<TextWebSocketFrame, TextWebSocketFrame> connection) -> connection.getInput()
                                 .map((TextWebSocketFrame webSocketFrame) -> webSocketFrame.text())))
-                .onErrorResumeNext(Observable.empty());
+                .onErrorResumeNext(Observable.empty())
+                .doOnCompleted(() -> {
+                  logger.info("Purging {} from job status cache", id);
+                  jobSchedulingInfoCache.remove(id);
+                })
+                .doOnUnsubscribe(() -> {
+                  logger.info("Purging {} from job status cache", id);
+                  jobSchedulingInfoCache.remove(id);
+                })
+                .doOnError((t) -> {
```

Comment:
noticed the onErrorResumeNext() up this observable chain, so this doOnError will not get executed, it should probably be before the onErrorResumeNext